### ISSUE TITLE
Crash on multi-file cuesheet

### ISSUE THREAD

**Issue by lnicola** [flacon/flacon#219]
I have a cue file that looks like this (with strings removed):

```
PERFORMER ""
TITLE ""
FILE "Title (A).flac" FLAC
  TRACK 01 AUDIO
    TITLE ""
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    TITLE ""
    INDEX 01 05:53:00
  TRACK 03 AUDIO
    TITLE ""
        INDEX 01 09:25:00
  TRACK 04 AUDIO
    TITLE ""
    PERFORMER ""
    INDEX 01 14:17:00
FILE "Title (B).flac" FLAC
  TRACK 05 AUDIO
    TITLE ""
    INDEX 01 00:00:00
  TRACK 06 AUDIO
    TITLE ""
    INDEX 01 04:37:00
  TRACK 07 AUDIO
    TITLE ""
    INDEX 01 10:37:00
  TRACK 08 AUDIO
    TITLE ""
    INDEX 01 16:15:00
```

and Flacon crashes (during conversion of track 3, which made it a little more confusing) with:

```
Warning: Splitter error for track  4 :  I can't read <b>Title (A).flac</b>:<br>Incorrect start or end time.
```

The indentation is a little broken, but fixing it doesn't help. What helps is removing one half of the file and converting the other. I didn't check, but I suspect it might be adding the length of `Title (A).flac` to the start and end times for tracks in `Title (B).flac`.

I can provide the files on request, but I imagine that some files filled with silence would cause the same issue.

**Comment by SokoloffA** [comment]
>I didn't check, but I suspect it might be adding the length of Title (A).flac to the start and end times for tracks in Title (B).flac.

No, the program processes the files separately. I suspect that the program mixed up the audio files for the first and second half of the disc. Please check it out.

![image](https://github.com/flacon/flacon/assets/854935/832406be-eac4-4eaa-b127-4a2cfa5d5529)

Anyway, I need to improve the audio file search.  Please send me the following information. 

1. Complete list of files in the directory. 
2. Duration of both audio files. You can get it using one of the following commands:  
     ```mediainfo AUDIO_FILE```
      ```ffmpeg -i  AUDIO_FILE```

**Comment by lnicola** [comment]
> I suspect that the program mixed up the audio files for the first and second half of the disc.

Yeah, it shows `Title (A).flac` twice, no `Title (B).flac`.

> Complete list of files in the directory.

`Title.cue`, `Title (A).flac`, `Title (B).flac`

> Duration of both audio files. You can get it using one of the following commands:

`ffprobe` output:

```
Input #0, flac, from 'Title (A).flac':
  Duration: 00:19:42.07, start: 0.000000, bitrate: 4965 kb/s
  Stream #0:0: Audio: flac, 192000 Hz, stereo, s32 (24 bit)

Input #0, flac, from 'Title (B).flac':
  Duration: 00:20:19.70, start: 0.000000, bitrate: 5010 kb/s
  Stream #0:0: Audio: flac, 192000 Hz, stereo, s32 (24 bit)
```

**Comment by SokoloffA** [comment]
I couldn't get the **A file twice**. But I was able to get **B file twice**,  I'll fix it and write about the results. 

How did you add  the file to the program?
* through "Add Folder"
* through "Add Disc" and click on the CUE file
* through "Add Disc" and click on "Title (A).flac"
* through "Add Disc" and click on "Title (B).flac"

**Comment by lnicola** [comment]
I used drag and drop, but both Add Disc (picking the cue) and Add Folder have the same effect. The later also tries to load `Title (B).flac` separately.

Anyway, I know this back and forth can be frustrating. I've uploaded the whole thing to https://nas.dend.ro/browse/public/st.zip (warning, it's 1.4 GB and probably not quite okay to distribute).

**Comment by SokoloffA** [comment]
With the real files, I reproduced the error. I will fix it.

**Comment by lnicola** [comment]
Fixed in 11.3.
